buildSrc/src/main/java/org/elasticsearch/test/NamingConventionsCheck.java

```diff
@@ -70,18 +72,21 @@ public static void main(String[] args) throws IOException {
                     checkMainClasses = true;
                     break;
                 case "--":
-                    rootPath = Paths.get(args[++i]);
+                    rootPathList = args[++i];
                     break;
                 default:
                     fail("unsupported argument '" + arg + "'");
             }
         }
 
         NamingConventionsCheck check = new NamingConventionsCheck(testClass, integTestClass);
-        if (checkMainClasses) {
-            check.checkMain(rootPath);
-        } else {
-            check.checkTests(rootPath, skipIntegTestsInDisguise);
+        for (String rootDir : rootPathList.split(Pattern.quote(File.pathSeparator))) {
```

Comment:
This seems inconsistent with the gradle code. There, a single dir is currently passed in. But here it looks like the path of that dir is split into its elements but then each read as a dir...

Reply:
I don't think we should do that. Each project should be checkable on its own. Combining all the sources into one task means to run precommit on a project requires running this one large task that looks at all test classes.